### PR TITLE
Clarify server player's watching position

### LINKED ISSUE #2178: ServerPlayerEntity#cameraPosition and others are misleading
I believe that the naming of the following attributes/methods/parameters are misleading:

1. `ServerPlayerEntity#cameraPosition` (and associated getter and setter)
2. `ServerChunkManager#updateCameraPosition`
3. `ThreadedAnvilChunkStorage#updateCameraPosition`
4. `ThreadedAnvilChunkStorage$EntityTracker#updateCameraPosition` (both)
5. The parameter `useCameraPosition` in `ThreadedAnvilChunkStorage#getChebyshevDistance`

Here's my reasoning:

Starting with `ServerPlayerEntity#cameraPosition`, the only time this is updated is within `ThreadedAnvilChunkStorage#updateCameraPosition`, where it is set to the `ChunkSectionPos` that the `ServerPlayerEntity` is currently in. This position is stored so that when the `ServerPlayerEntity` changes `ChunkSectionPos`s, chunks within the range of this but not within the range of the `ServerPlayerEntity`'s current chunk position can be unloaded (this is my understanding). It isn't related to `ServerPlayerEntity#cameraEntity`, only the `ServerPlayerEntity` itself.

Next with `ServerChunkManager#updateCameraPosition`, `ThreadedAnvilChunkStorage#updateCameraPosition`, and both `ThreadedAnvilChunkStorage$EntityTracker#updateCameraPosition`s; the only argument given to these methods is a `ServerPlayerEntity`. Within the methods, the `ServerPlayerEntity#cameraEntity` is never queried. Instead, the `ServerPlayerEntity`'s chunk position is used (or in the case of `ThreadedAnvilChunkStorage#updateCameraPosition` where `ServerPlayerEntity#cameraPosition` is also used... see above regarding that attribute).

Finally with the `useCameraPosition` parameter in `ThreadedAnvilChunkStorage#getChebyshevDistance`, this boolean is used to determine if `ServerPlayerEntity#cameraPosition` should be used instead of the `ServerPlayerEntity`'s current chunk position. If `ServerPlayerEntity#cameraPosition` is renamed, this should be also.

I feel that names for these attributes/methods/parameters would be better if they were focused around the player instead of some "camera." While it could be thought that the camera referenced in these places is more of a "server camera" where chunk loading and unloading takes place, I still feel that the name conflicts with `ServerPlayerEntity#cameraEntity` as it reads that the position of this and `ServerPlayerEntity#cameraPosition` should be equivalent, but this is not the case.

I'd like to become more involved with the development of Yarn mappings, so I wouldn't mind submitting a PR with different names, but I'd rather get the community's feedback on this first. Also, if there are other attributes/methods/parameters within this part of the code that bear similar names and I missed, please let me know. Thanks!

### PR DESCRIPTION
**Pull request by liach**
Fixes #2178

Signed-off-by: liach <liach@users.noreply.github.com>

### CODE REVIEW

**Comment by liach** [comment]
Pinging @The-PPWD for review.

**Comment by The-PPWD** [comment]
I like it sans a few things:

1. I think `watchingSection` could benefit by being named `lastWatchedSection` to distinguish it from being the `ServerPlayerEntity`'s current `ChunkSectionPos` (i.e. `watchingSection` != `ChunkSectionPos.from(player)`).

2. `fetchWatchingSection` sounds too similar to `getWatchingSection` (without reading the docs of course). Perhaps `updateWatchingSection` (or `updateLastWatchedSection` given point 1). In both places where this method is called, the return value is ignored anyway.

Otherwise, I really like the changes and they help to distinguish the chunk loading/unloading method names and the entity tracking method names which will help make things a lot clearer.

**Comment by liach** [comment]
> 
> 
> I like it sans a few things:
> 
>     1. I think `watchingSection` could benefit by being named `lastWatchedSection` to distinguish it from being the `ServerPlayerEntity`'s current `ChunkSectionPos` (i.e. `watchingSection` != `ChunkSectionPos.from(player)`).

Hmm, now I think we can unify the names to `watchedSection` and `trackedSection`. `ed` already indicates `last`, and this section pos might be still valid.

> 
>     2. `fetchWatchingSection` sounds too similar to `getWatchingSection` (without reading the docs of course). Perhaps `updateWatchingSection` (or `updateLastWatchedSection` given point 1). In both places where this method is called, the return value is ignored anyway.
> 

Yeah, I originally didn't name it `updatePosition` as I didn't realize it updates the position for both watching and tracking systems, and forgot to give `fetchWatchingPosition` a better name.

> 
> Otherwise, I really like the changes and they help to distinguish the chunk loading/unloading method names and the entity tracking method names which will help make things a lot clearer.

Thanks!

**Review by The-PPWD — APPROVED** [review]
I like it!

File: mappings/net/minecraft/server/world/ThreadedAnvilChunkStorage.mapping [hunk 1]
```diff
@@ -91,7 +91,11 @@ CLASS net/minecraft/class_3898 net/minecraft/server/world/ThreadedAnvilChunkStor
 	METHOD method_18704 getSquaredDistance (Lnet/minecraft/class_1923;Lnet/minecraft/class_1297;)D
 		ARG 0 pos
 		ARG 1 entity
-	METHOD method_18713 updateCameraPosition (Lnet/minecraft/class_3222;)V
+	METHOD method_18713 updatePosition (Lnet/minecraft/class_3222;)V
+		COMMENT Updates the chunk section position the {@code player}. This updates the player
```
Comment (YanisBft) [inline mappings/net/minecraft/server/world/ThreadedAnvilChunkStorage.mapping]:
```suggestion
		COMMENT Updates the chunk section position of the {@code player}. This updates the player
```

File: mappings/net/minecraft/server/world/ThreadedAnvilChunkStorage.mapping [hunk 2]
```diff
@@ -125,30 +135,54 @@ CLASS net/minecraft/class_3898 net/minecraft/server/world/ThreadedAnvilChunkStor
...
+			ARG 2 entity
+			ARG 3 maxDistance
+			ARG 4 tickInterval
+			ARG 5 alwaysUpdateVelocity
 		METHOD equals (Ljava/lang/Object;)Z
 			ARG 1 o
 		METHOD method_18728 stopTracking ()V
-		METHOD method_18729 updateCameraPosition (Ljava/util/List;)V
+		METHOD method_18729 updateTrackedStatus (Ljava/util/List;)V
+			COMMENT Update the tracked status of this tracker' entity for the given players.
```
Comment (YanisBft) [inline mappings/net/minecraft/server/world/ThreadedAnvilChunkStorage.mapping]:
```suggestion
			COMMENT Updates the tracked status of this tracker's entity for the given players.
```

File: mappings/net/minecraft/server/world/ThreadedAnvilChunkStorage.mapping [hunk 3]
```diff
@@ -125,30 +135,54 @@ CLASS net/minecraft/class_3898 net/minecraft/server/world/ThreadedAnvilChunkStor
...
 			ARG 1 players
 		METHOD method_18730 sendToOtherNearbyPlayers (Lnet/minecraft/class_2596;)V
 			ARG 1 packet
 		METHOD method_18733 stopTracking (Lnet/minecraft/class_3222;)V
 			ARG 1 player
 		METHOD method_18734 sendToNearbyPlayers (Lnet/minecraft/class_2596;)V
 			ARG 1 packet
-		METHOD method_18736 updateCameraPosition (Lnet/minecraft/class_3222;)V
+		METHOD method_18736 updateTrackedStatus (Lnet/minecraft/class_3222;)V
+			COMMENT Update the tracked status of this tracker's entity for the {@code player}.
```
Comment (YanisBft) [inline mappings/net/minecraft/server/world/ThreadedAnvilChunkStorage.mapping]:
```suggestion
			COMMENT Updates the tracked status of this tracker's entity for the {@code player}.
```

**Review by YanisBft — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**